### PR TITLE
Profile: add ability to edit profile info

### PR DESCRIPTION
**Pull request by ezhivitsa**
@anorsich Please, review

### CODE REVIEW

File: src/client/components/profile/profile.jsx [hunk 1]
```diff
@@ -27,32 +33,37 @@ class Profile extends React.Component {
...
+      email: '',
       errors: {},
     };
 
     this.updateUser = this.updateUser.bind(this);
   }
 
+  async componentDidMount() {
+    try {
+      await this.props.fetchUser('current');
```
Comment (anorsich) [inline src/client/components/profile/profile.jsx]: I know @IharKrasnik probably will like this as well, but I strongly believe that errors must be used for unexpected errors. In this case, we expect that API could return 500, or 404. I would better return an object with `isBadRequest` and `data` fields, so in components, we can check and handle bad requests. We could handle some errors generally (like for 500 — show notification, for 404 send to the not found page)

**Comment by anorsich** [comment]
@umitskevich @IharKrasnik Let me know what you guys think about this PR :).

File: src/client/components/profile/profile.jsx (hunk 1 above)
Comment (ezhivitsa) [reply src/client/components/profile/profile.jsx]: @anorsich `isBadRequest` will help us to handle only 'good' errors (validation errors or some another errors that we can expect). But in the same time we can get some another errors (that server is not available or another 'bad' error), therefore to handle both 'good' and 'bad' errors we should have if statement with `isBadRequest` and `try - catch` construction.
I think that will be better use only `try - catch` construction to handle all possible errors.

Comment (anorsich) [reply src/client/components/profile/profile.jsx]: @ezhivitsa Actually, any kind of response from the server is expected behaviour (404,403, 500, etc) and UI should handle it. We no need to catch wrap all unexpected behaviours and the errors need to be thrown and logged in a central way. We do this at Maqpie just by introducing global error handler and sending errors to the server (ELK stack). For now we could just keep unexpected errors as is without server logging.

**Pull request merged**